mpas_analysis/ocean/histogram.py

```diff
@@ -0,0 +1,589 @@
+        write_netcdf(ds_weights, weights_filename)
+
+
+class PlotRegionHistogramSubtask(AnalysisTask):
+    """
+    Plots a histogram diagram for a given ocean region
+
+    Attributes
+    ----------
+    regionGroup : str
+        Name of the collection of region to plot
+
+    regionName : str
+        Name of the region to plot
+
+    sectionName : str
+        The section of the config file to get options from
+
+    controlConfig : mpas_tools.config.MpasConfigParser
+        The configuration options for the control run (if any)
+
+    mpasClimatologyTask : ``MpasClimatologyTask``
+        The task that produced the climatology to be remapped and plotted
+
+    mpasMasksSubtask : ``ComputeRegionMasksSubtask``
+        A task for creating mask MPAS files for each region to plot, used
+        to get the mask file name
+
+    obsDicts : dict of dicts
+        Information on the observations to compare against
+
+    variableList: list of str
+        list of variables to plot
+
+    season : str
+        The season to compute the climatology for
+    """
+
+    def __init__(self, parentTask, regionGroup, regionName, controlConfig,
+                 sectionName, fullSuffix, mpasClimatologyTask,
+                 mpasMasksSubtask, obsMasksSubtask, obsDicts, variableList,
+                 weightList, season):
+
+        """
+        Construct the analysis task.
+
+        Parameters
+        ----------
+        parentTask :  ``AnalysisTask``
+            The parent task, used to get the ``taskName``, ``config`` and
+            ``componentName``
+
+        regionGroup : str
+            Name of the collection of region to plot
+
+        regionName : str
+            Name of the region to plot
+
+        controlconfig : mpas_tools.config.MpasConfigParser, optional
+            Configuration options for a control run (if any)
+
+        sectionName : str
+            The config section with options for this regionGroup
+
+        fullSuffix : str
+            The regionGroup and regionName combined and modified to be
+            appropriate as a task or file suffix
+
+        mpasClimatologyTask : ``MpasClimatologyTask``
+            The task that produced the climatology to be remapped and plotted
+
+        mpasMasksSubtask : ``ComputeRegionMasksSubtask``
+            A task for creating mask MPAS files for each region to plot, used
+            to get the mask file name
+
+        obsDicts : dict of dicts
+            Information on the observations to compare agains
+
+        season : str
+            The season to comput the climatogy for
+        """
+
+        # first, call the constructor from the base class (AnalysisTask)
+        super(PlotRegionHistogramSubtask, self).__init__(
+            config=parentTask.config,
+            taskName=parentTask.taskName,
+            componentName=parentTask.componentName,
+            tags=parentTask.tags,
+            subtaskName=f'plot{fullSuffix}_{regionName}_{season}')
```

Comment:
Similarly to above, I think this might be better with `regionGroupSuffix` instead of `fullSuffix`:
```suggestion
            subtaskName=f'plot_{regionGroupSuffix}_{regionName}_{season}')
```